### PR TITLE
[MP-463] Add types used in other projects

### PR DESCRIPTION
**Pull request by cactusaurelius**
### :gem: Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### :scroll: Example code
```js

``` 

## :vertical_traffic_light: How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## :checkered_flag: Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] **I have added tests that prove my fix is effective or that my feature works**
- [ ] **New and existing unit tests pass locally with my changes**
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Review by HazemKhaled — CHANGES_REQUESTED** [review]

File: src/invoice.ts [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+  isInclusiveTax: boolean;
+  items?: InvoiceItems[];
+  lastModifiedTime: string;
+  last_modified_time: string;
+  message: string;
+  omsId: string;
+  referenceNumber: string;
+  reference_number: string;
+  shippingCharge: number;
+  shipping_charge: number;
```
Comment (HazemKhaled) [inline src/invoice.ts:85]: Why `shippingCharge` and `shipping_charge`?

fix it please

File: src/invoice.ts [hunk 2]
```diff
@@ -0,0 +1,186 @@
+/**
+ * Invoice type
+ *
+ * @export
+ * @interface Invoice
+ */
+export interface InvoiceResponse {
+  adjustment: number;
```
Comment (HazemKhaled) [inline src/invoice.ts:8]: It should be as same as invoice, no need to have 2 types

File: src/invoice.ts [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+  useSavedPaymentMethods: string;
+}
+
+/**
+ * OMS Invoice Apply credit response
+ *
+ * @export
+ * @interface OmsApplyCreditResponse
+ */
+export interface OmsApplyCreditResponse {
```
Comment (HazemKhaled) [inline src/invoice.ts:124]: Name should be serviceActionNameParamas or serviceActioNameResponse

ex: invoiceGetParams

File: src/payment.ts [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  amountApplied: number;
+  invoiceId: string;
+}
+
+/**
+ * PaymentInvoiceResponse
+ *
+ * @interface PaymentInvoiceResponse
+ */
+export interface PaymentInvoiceResponse {
```
Comment (HazemKhaled) [inline src/payment.ts:61]: make it serviceActionNameResponse

File: src/payment.ts [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+  purchase_units: {
+    amount: {
+      currency_code: string;
+      value: number;
+    };
+    description?: string;
+  }[];
+  reference: string;
+  referenceNumber?: string;
+  reference_number?: string;
```
Comment (HazemKhaled) [inline src/payment.ts:97]: underscore or camel case?

File: src/sales-order/order.ts [hunk 6]
```diff
@@ -37,6 +37,7 @@ export interface SalesOrder {
   shipmentDate: Date;
   shipping: Address;
   shippingCharge: number;
+  shipping_charge: number;
```
Comment (HazemKhaled) [inline src/sales-order/order.ts:40]: duplicated

**Review by HazemKhaled — CHANGES_REQUESTED** [review]

File: src/payment.ts (hunk 5 above)
Comment (cactusaurelius) [reply src/payment.ts:97]: response from OMS is camel case then sanitized into snake case
https://github.com/Knawat/mp/blob/9fa9d4b07ad7596fddbda025eb5afb86871fd939/services/invoices.service.ts#L290

**Pull request closed without merging**